server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2473,12 +2473,15 @@ private IndexWriterConfig getIndexWriterConfig() {
                 iwc.setMaxFullFlushMergeWaitMillis(maxFullFlushMergeWaitMillis);
                 mergePolicy = new MergeOnFlushMergePolicy(mergePolicy);
             } else {
+                iwc.setMaxFullFlushMergeWaitMillis(0);
                 logger.warn(
                     "The {} is enabled but {} is set to 0, merge on flush will not be activated",
                     IndexSettings.INDEX_MERGE_ON_FLUSH_ENABLED.getKey(),
                     IndexSettings.INDEX_MERGE_ON_FLUSH_MAX_FULL_FLUSH_MERGE_WAIT_TIME.getKey()
                 );
             }
+        } else {
+            iwc.setMaxFullFlushMergeWaitMillis(0);
```

Comment:
I think we want to retain the `500ms` default here per the [commit message](https://github.com/apache/lucene/commit/b5795db0cf517f8942eed868752249df9b105603)?

Reply:
I updated the [testcase to 0ms](11040996627992fd994496f4ce37db8e58c63e78), though, so we can retain the behavior of not exposing aborted docs.

Reply:
No, we explicitly disable merge on refresh since the setting is set to `false`